Robot.java_03_14_17

```diff
@@ -1,418 +1 @@
-
-		//switching between drive modes - Halie 
-		
-		driveMode ="TANK";
-		
-		cameraMode="BACK";
-		//if a is pressed, driveMode = "ARCADE"
-		//if b is pressed, diveMode= "TANK"
-		//togglewhenpressed
-		TriggerButton.whenPressed(new SetTankDrive());
-		SideButton.whenPressed(new SetArcadeDrive());
-		RightButton.whenPressed(new SetTankDriveBack());
-		
-		HighButton.whenPressed(new FrontCamera());
-		LowButton.whenPressed(new BackCamera());
-		//TopButton.whenPressed(new GreenLight());
-		
-		//RightButton.whenPressed(new MoveForward());
-		
-		
-		state = "forward1";
-		
-	}
-
-	
-	
-	/**
-	 * This function is called once each time the robot enters Disabled mode.
-	 * You can use it to reset any subsystem information you want to clear when
-	 * the robot is disabled.
-	 */
-	@Override
-	public void disabledInit() {
-		this.left.disable();
-		this.right.disable();
-
-	}
-
-	@Override
-	public void disabledPeriodic() {
-		Scheduler.getInstance().run();
-	}
-
-	/**
-	 * This autonomous (along with the chooser code above) shows how to select
-	 * between different autonomous modes using the dashboard. The sendable
-	 * chooser code works with the Java SmartDashboard. If you prefer the
-	 * LabVIEW Dashboard, remove all of the chooser code and uncomment the
-	 * getString code to get the auto name from the text box below the Gyro
-	 *
-	 * You can add additional auto modes by adding additional commands to the
-	 * chooser code above (like the commented example) or additional comparisons
-	 * to the switch structure below with additional strings & commands.
-	 */
-	@Override
-	public void autonomousInit() {
-		autonomousCommand = chooser.getSelected();
-		
-		
-		double KpR=9/72.0;
-		double KpL=9/72.0;
-		double Ki = 0.01/72.0;
-		double Kd= 1/72.0;
-		//double KfR= 1.4/36.0;
-		//double KfL=1.45/36.0;
-		double Kf = 1.0;
-		
-		this.right = new Good_DriveSide(KpR, Ki, Kd, Kf, rightEncoder, rightSpark);
-		right.setInverted(true);
-		this.left = new Good_DriveSide(KpL, Ki, Kd, Kf, leftEncoder, leftSpark);
-
-		
-		driveTrain = new RobotDrive(left,right);
-
-		/*
-		 * String autoSelected = SmartDashboard.getString("Auto Selector",
-		 * "Default"); switch(autoSelected) { case "My Auto": autonomousCommand
-		 * = new MyAutoCommand(); break; case "Default Auto": default:
-		 * autonomousCommand = new ExampleCommand(); break; }
-		 */
-		//autonomousCommand = new MoveForward();
-		// schedule the autonomous command (example)
-		if (autonomousCommand != null)
-			autonomousCommand.start();
-		
-
-		myTimer.reset();
-		myTimer.start();
-		
-		
-		System.out.println("Autonomous!");
-		System.out.println("StartTime: " + myTimer.get());
-		System.out.println();
-		System.out.println();
-		
-	//leftEncoder.reset();
-		gyro.calibrate();
-//		System.out.println("gyro calibration: " + gyro.getAngle());
-		
-		
-		
-	}
-
-	/**
-	 * This function is called periodically during autonomous
-	 */
```

Comment:
For autonomous mode, given the time constraint, your rough approach here is good. It's generally what we call "dead reckoning" because it's not using a sensor to know if it's close to the target. However, you are using the encoders and gyro to give you an idea of where you are relative to your start location. This is WAY better than saying "turn on motors for 0.3s, etc). You'll have to tune the distances on the practice field, but you should be able to get it to get it to pass the line 95% of the time and get a gear on from a position or two maybe 50% of the time. 

So, I would recommend restructuring the code slightly so that you have a few different "programs" that you can run depending on your starting location, but use your general approach as you've done already. I beleive Anya knows how to set a variable from the control panel, which you can use to toggle between your programs, but if not, you can just recompile the code to change one variable.

Specifically, if you can have a class field called something like autonomousProgram that's either an enum or a string. You set the value of that variable in autonomousInit() either hard coding, or from reading from the console. e.g.

autonomousProgram = "middleDepositGear"

Then, in autonomousPeriodic, you branch based on that variable. e.g.

if (autonomousProgram == "middleDespoitGear") {
   // Run the steps to get the robot to deposit the gear from the middle position
elif (autonomousProgram == "sprintPastLine" ) {
   // Run the steps that cause the robot to get past the distance line
}
...